packages/compass-preferences-model/index.d.ts

```diff
@@ -1,15 +1,29 @@
-declare class CompassPreferencesModel {
-  fetch: () => void;
-  save: () => void;
-  set: (
-    key: string,
-    value: unknown
-  ) => void;
-};
-export default CompassPreferencesModel;
-
 export enum THEMES {
   DARK = 'DARK',
   LIGHT = 'LIGHT',
   OS_THEME = 'OS_THEME'
-};
+}
+
+export type UserPreferences = {
+  showedNetworkOptIn: boolean; // Has the settings dialog has been shown before.
+  autoUpdates: boolean;
+  enableMaps: boolean;
+  trackErrors: boolean;
+  trackUsageStatistics: boolean;
+  enableFeedbackPanel: boolean;
+  theme: THEMES;
+}
+
+declare class CompassPreferencesModel {
+  fetchPreferences: () => Promise<void>;
+  savePreferences: (preferences: any) => Promise<void>;
+  getPreferenceValue: any;
+  onPreferenceChanged: (preferenceName: string, callback: () => void) => void;
```

Comment:
```suggestion
  onPreferenceChanged: (preferenceName: keyof UserPreferences, callback: () => void) => void;
```

should there be a way to unsubscribe?

Reply:
I was thinking about it. But when and where unsubscribing should happen? I looks at other similar events in Compass and didn't find a pattern how we do that.